[inductor] correctly generate grid info for benchmark_kernel

chenyang78 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #118202

Previously, we generated the grid argument with tree.numel for
a benchmark TritonKernel. This was not correct, because it
didn't match the launch config used for profiling and running.

This PR fixed the issue by emitting the grid value computed
by the kernel's grid_fn, which is used by the profiler and
the kernel's runner.

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @peterbell10 @ipiszy @yf225 @kadeng @muchulee8 @aakhundov @ColinPeppler

Commit:
[inductor] correctly generate grid info for benchmark_kernel

Previously, we generated the grid argument with tree.numel for
a benchmark TritonKernel. This was not correct, because it
didn't match the launch config used for profiling and running.

This PR fixed the issue by emitting the grid value computed
by the kernel's grid_fn, which is used by the profiler and
the kernel's runner.

[ghstack-poisoned]

shunting314 commented on test/inductor/test_kernel_benchmark.py:
```diff
@@ -51,9 +52,18 @@ def verify_compiled_kernels(self):
+            with open(compiled_module.__file__) as f:
+                source_code = f.read()
+            FileCheck().check_count(
+                "grid=(608, 1, 1)",
```
I bet this test is potentially quite flaky since the grid size depends on the config we pick for triton matmul template which is decided by the autotuning result.

We probably can either read back the block size from the generated wrapper code and use that to compute the grid, or mock the config list to a single config so we know for sure what grid size should be.

chenyang78 replied:
Yeah, agreed. Fixed. Thanks.

shunting314 reviewed (approved):
Thanks for fixing the grid size computation for the benchmark harness of triton templates.

Commit:
Update on "[inductor] correctly generate grid info for benchmark_kernel"


Previously, we generated the grid argument with tree.numel for
a benchmark TritonKernel. This was not correct, because it
didn't match the launch config used for profiling and running.

This PR fixed the issue by emitting the grid value computed
by the kernel's grid_fn, which is used by the profiler and
the kernel's runner.

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx peterbell10 ipiszy yf225 kadeng muchulee8 aakhundov ColinPeppler

[ghstack-poisoned]

chenyang78 commented:
@pytorchbot merge
